sdk/core/core/test/cmocka/test_az_http.c

```diff
@@ -145,6 +147,31 @@ static void test_http_request(void** state)
         az_span_ptr(AZ_SPAN_FROM_STR("https://antk-keyvault.vault.azure.net/secrets/Password/"
                                      "path?api-version=7.0&test-param=token")));
   }
+  {
+    uint8_t buf[100];
+    uint8_t header_buf[(2 * sizeof(az_pair))];
+    memset(buf, 0, sizeof(buf));
+    memset(header_buf, 0, sizeof(header_buf));
+
+    az_span url_span = AZ_SPAN_FROM_BUFFER(buf);
+    az_span remainder = az_span_copy(url_span, hrb_url);
+    assert_int_equal(az_span_size(remainder), 100 - az_span_size(hrb_url));
+    az_span header_span = AZ_SPAN_FROM_BUFFER(header_buf);
+    _az_http_request hrb;
+
+    TEST_EXPECT_SUCCESS(az_http_request_init(
+        &hrb,
+        &az_context_app,
+        az_http_method_get(),
+        url_span,
+        az_span_size(hrb_url),
+        header_span,
+        AZ_SPAN_FROM_STR("body")));
+
+    // Empty header
+    assert_return_code(
+        az_http_request_append_header(&hrb, AZ_SPAN_FROM_STR("header"), AZ_SPAN_NULL), AZ_OK);
```

Comment:
Since we are now allowing `AZ_SPAN_NULL` as the value in the key:value `az_pair`, do we need more changes in `az_http_request_append_header` on the internal representation using `az_pair` and how we increase the offset? Should we not use `az_pair` anymore for this if one of the spans are optional? I don't know if increasing by `sizeof(az_pair)` is correct if we are only adding a single header span, with no value. Can you make sure the offsets are done correctly?

Reply:
yeah, we still need az_pair.  It is the same size to have  one `az_span` or `AZ_SPAN_NULL`.  And the trouble of managing not adding the value when it is null is high!  compared to what we would win.
For instance, for looping headers, we would need a separator to know where each header end (instead of fixed size of az_pair)